lib/ansible/module_utils/k8svirt/common.py

```diff
@@ -72,32 +73,60 @@ def __configure_by_file(self):
 
         try:
             if not self.params.get('verify_ssl'):
-                sdk.configuration.verify_ssl = False
-            kube_config.load_kube_config(
-                config_file=config_file,
-                context=self.params.get('context'),
-                client_configuration=sdk.configuration
-            )
-            return sdk.DefaultApi()
+                verify_ssl = False
+            else:
+                verify_ssl = self.params.get('verify_ssl')
+
+            kubevirt_client = self.__create_kubevirt_client(
+                config_file, verify_ssl, self.params.get('context'))
+            core_client = self.__create_core_client(
+                config_file, verify_ssl, self.params.get('context'))
+            return kubevirt_client, core_client
         except (IOError, ConfigException):
             raise
 
+    def __create_core_client(self, config_file, verify_ssl, context):
+        configuration = Configuration()
+        configuration.verify_ssl = verify_ssl
+        Configuration.set_default(configuration)
+        kube_config.load_kube_config(config_file=config_file)
+        if context:
+            return core_client.CoreV1Api(
+                api_client=config.new_client_from_config(context=context))
+
+        return core_client.CoreV1Api()
+
+    def __create_kubevirt_client(self, config_file, verify_ssl, context):
+        sdk.configuration.verify_ssl = verify_ssl
+        kube_config.load_kube_config(
+            config_file=config_file,
+            context=context,
+            client_configuration=sdk.configuration
+        )
+        return sdk.DefaultApi()
+
     def __configure_by_params(self):
         """ Return API client from configuration file """
         auth_args = AUTH_ARG_SPEC.keys()
+        core_configuration = Configuration()
 
         for key, value in iteritems(self.params):
```

Comment:
Would be good to extract is as a method which would accept configuration as param.

Reply:
We can improve this later